ui/components/connections/index.js

```diff
@@ -150,7 +158,77 @@ function Connections({ classes, updateProgress, /*onOpenCreateConnectionModal,*/
   const { notify } = useNotification();
   const StyleClass = useStyles();
 
-  // const statuses = ['ignored', 'connected', 'REGISTERED', 'discovered', 'deleted'];
+  const statuses = ['ignored', 'connected', 'REGISTERED', 'discovered', 'deleted'];
+
+  const status = (value) => {
+    switch (value) {
+      case 'ignored':
+        return (
+          <MenuItem value={value}>
+            <Chip
+              className={classNames(classes.statusChip, classes.ignored)}
+              avatar={<RemoveCircleIcon />}
+              label={value}
+            />
+          </MenuItem>
+        );
+      case 'connected':
+        return (
+          <MenuItem value={value}>
+            <Chip
+              className={classNames(classes.statusChip, classes.connected)}
+              value={value}
+              avatar={<CheckCircleIcon />}
+              label={value}
+            />
+          </MenuItem>
+        );
+      case 'REGISTERED':
+        return (
+          <MenuItem value={value}>
+            <Chip
+              className={classNames(classes.statusChip, classes.registered)}
+              value={value}
+              avatar={<AssignmentTurnedInIcon />}
+              label={value.toLowerCase()}
+            />
+          </MenuItem>
+        );
+      case 'discovered':
+        return (
+          <MenuItem value={value}>
+            <Chip
+              className={classNames(classes.statusChip, classes.discovered)}
+              value={value}
+              avatar={<ExploreIcon />}
+              label={value}
+            />
+          </MenuItem>
+        );
+      case 'deleted':
+        return (
+          <MenuItem value={value}>
+            <Chip
+              className={classNames(classes.statusChip, classes.deleted)}
+              value={value}
+              avatar={<DeleteForeverIcon />}
+              label={value}
+            />
+          </MenuItem>
+        );
+      default:
+        return (
+          <MenuItem value={value}>
+            <Chip
+              className={classNames(classes.statusChip, classes.discovered)}
+              value={value}
+              avatar={<ExploreIcon />}
+              label={value}
+            />
+          </MenuItem>
+        );
+    }
+  };
```

Comment:
@sudhanshutech  if you would create a single status component that would map over STATUS enum . it will be more maintainable

Reply:
updated to map on states defined in enum 👍